src/test/java/org/sonarsource/sonarlint/ls/mediumtests/AbstractLanguageServerMediumTests.java

```diff
@@ -117,7 +121,7 @@ static void startServer() throws Exception {
 
     client = new FakeLanguageClient();
 
-    var executor = Executors.newSingleThreadExecutor();
+    var executor = Executors.newFixedThreadPool(2);
```

Comment:
What is the reason for this change?

Reply:
`ServerMain` will wait for a shutdown and therefore, I needed a second thread that starts the main, see `executor.submit(() -> cmd.execute(clonedArgs));` down below. If the test code here uses `newSingleThreadExecutor()`, the `clientSideLauncher` and `cmd` are dead locked.

That is necessary due to the adjustments in `ServerMain` and `SonarLintLanguageServer`. The adjusted structure of `ServerMain` and `SonarLintLanguageServer` should also provide the ability to fix the potential resource leak:

![grafik](https://user-images.githubusercontent.com/16558417/219451364-5c91bbfc-d6b9-489c-af79-facd7d80e54e.png)